### PR TITLE
stub: update ServerCalls to include status messages

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Additionally:

* Prefer RuntimeException from status
* remove redundant "static" on interfaces
* use checkState to shorten code

### CODE REVIEW

File: stub/src/main/java/io/grpc/stub/ServerCalls.java [hunk 1]
```diff
@@ -413,7 +406,7 @@ public static void asyncUnimplementedUnaryCall(MethodDescriptor<?, ?> methodDesc
     responseObserver.onError(Status.UNIMPLEMENTED
         .withDescription(String.format("Method %s is unimplemented",
             methodDescriptor.getFullMethodName()))
-        .asException());
+        .asRuntimeException());
```
Comment (dapengzhang0) [inline stub/src/main/java/io/grpc/stub/ServerCalls.java]: What's the difference here? Should we always onError SRE?

Comment (carl-mastrangelo) [reply stub/src/main/java/io/grpc/stub/ServerCalls.java]: I prefer SRE

**Comment by dapengzhang0** [comment]
PS: better put reformat changes in a separate commit for review

**Comment by carl-mastrangelo** [comment]
Changes that are too small for single commits are not a good idea IMO.  Especially since there is at least a day between sending out a PR and it actually getting reviewed.

**Comment by ejona86** [comment]
> Changes that are too small for single commits are not a good idea IMO.

But having a 3 line change become 30 is noisy and the real change gets lost in the noise.

> Especially since there is at least a day between sending out a PR and it actually getting reviewed.

Just send them both out at the same time. No additional latency.

**Pull request merged**